homepage/homepage/app/(docs)/docs/[framework]/[...slug]/authentication/overview.mdx

```diff
@@ -0,0 +1,404 @@
+  <ClerkProvider publishableKey={PUBLISHABLE_KEY} afterSignOutUrl="/">
+    <JazzProvider>
+      <App />
+    </JazzProvider>
+  </ClerkProvider>
+);
+```
+</CodeGroup>
+</ContentByFramework>
+<ContentByFramework framework="react-native">
+<CodeGroup>
+```tsx
+import { JazzProviderWithClerk } from "jazz-react-native-auth-clerk";
+
+function JazzAndAuth({ children }: { children: React.ReactNode }) {
+  const clerk = useClerk();
+
+  return (
+    <JazzProviderWithClerk
+      clerk={clerk}
+      sync={{
+        peer: `wss://cloud.jazz.tools/?key=${apiKey}`,
+      }}
+    >
+      {children}
+    </JazzProviderWithClerk>
+  );
+}
+
+export default function RootLayout() {
+  const publishableKey = process.env.EXPO_PUBLIC_CLERK_PUBLISHABLE_KEY;
+
+  if (!publishableKey) {
+    throw new Error(
+      "Missing Publishable Key. Please set EXPO_PUBLIC_CLERK_PUBLISHABLE_KEY in your .env",
+    );
+  }
+
+  return (
+    <ClerkProvider tokenCache={tokenCache} publishableKey={publishableKey}>
+      <ClerkLoaded>
+        <JazzAndAuth>
+          <Slot />
+        </JazzAndAuth>
+      </ClerkLoaded>
+    </ClerkProvider>
+  );
+}
+```
+</CodeGroup>
+</ContentByFramework>
+
+<ContentByFramework framework="react">
+Then you can use the [Clerk auth methods](https://clerk.com/docs/references/react/overview) to log in and sign up:
+<CodeGroup>
+{/* prettier-ignore */}
+```tsx
+import { SignInButton } from "@clerk/clerk-react";
+import { useAccount, useIsAuthenticated } from "jazz-react";
+
+export function AuthButton() {
+  const { logOut } = useAccount();
+
+  const isAuthenticated = useIsAuthenticated();
+
+  if (isAuthenticated) {
+    return <button onClick={() => logOut()}>Logout</button>;
+  }
+
+  return <SignInButton />;
+}
+```
+</CodeGroup>
+</ContentByFramework>
+
+<ContentByFramework framework="react-native">
+Then you can use the [Clerk auth methods](https://clerk.com/docs/references/expo/overview) to log in and sign up:
+<CodeGroup>
+{/* prettier-ignore */}
+```tsx
+import { SignInButton } from "@clerk/clerk-expo";
+import { useAccount, useIsAuthenticated } from "jazz-react-native";
+
+export function AuthButton() {
+  const { logOut } = useAccount();
+  const { signIn, setActive, isLoaded } = useSignIn();
+
+  const isAuthenticated = useIsAuthenticated();
+
+  if (isAuthenticated) {
+    return <button onClick={() => logOut()}>Logout</button>;
+  }
+
+  // Login code with Clerk Expo
+}
+```
+</CodeGroup>
+</ContentByFramework>
+</ContentByFramework>
+
+## Migrating data from anonymous to authenticated account
+
+You may want allow your users to use your app without authenticating (a poll response for example). But they might make changes that they'd like to keep when they do authenticate.
+
+When they log in, the anonymous account will be deleted and this could lead to unexpected data loss.
+
+To avoid this situation we provide the `onAnonymousAccountDiscarded` handler to migrate the data from the anonymous account to the authenticated one.
```

Comment:
```suggestion
To avoid this situation we provide the `onAnonymousAccountDiscarded` handler to migrate the data from the anonymous account to the existing authenticated one.
```